### ISSUE TITLE
packer did not remove VirtualBox virtual machine when 'Typing the boot command' fails

### ISSUE THREAD

**Issue by pstrzelczak** [hashicorp/packer#5094]
When 'Typing the boot command' phase fails and VM enters 'Guru meditation' phase VM is not properly cleaned up. I was using very simple json template to set up CentOS (https://gist.github.com/pstrzelczak/1d679e15e1ed419956d235d08530accb) and launch it in a loop via

`packer build -var vm_name=foo -var no_gui=true centos-vbox.json`

4th iteration failed with that error. I will try to reproduce this issue with packer debug logs enabled. 

```Packer v1.0.0```

```
00:00:00.013764 OS Product: Linux
00:00:00.013764 OS Release: 4.4.75-1.el7.elrepo.x86_64
00:00:00.013765 OS Version: #1 SMP Thu Jun 29 10:01:58 EDT 2017
00:00:00.013834 DMI Product Name: KVM
00:00:00.013841 DMI Product Version: RHEL 7.3.0 PC (i440FX + PIIX, 1996)
00:00:00.013911 Host RAM: 7983MB (7.7GB) total, 7752MB (7.5GB) available
```

```
virtualbox-iso output will be in this color.

==> virtualbox-iso: Downloading or copying Guest additions
    virtualbox-iso: Downloading or copying: file:///usr/share/virtualbox/VBoxGuestAdditions.iso
==> virtualbox-iso: Downloading or copying ISO
    virtualbox-iso: Downloading or copying: file:///iso/CentOS-7-x86_64-Minimal-1611.iso
    virtualbox-iso: Error downloading: stat /iso/CentOS-7-x86_64-Minimal-1611.iso: no such file or directory
    virtualbox-iso: Downloading or copying: http://mirror.hh.se/centos/7.3.1611/isos/x86_64/CentOS-7-x86_64-Minimal-1611.iso
==> virtualbox-iso: Starting HTTP server on port 8625
==> virtualbox-iso: Creating virtual machine...
==> virtualbox-iso: Creating hard drive...
==> virtualbox-iso: Creating forwarded port mapping for communicator (SSH, WinRM, etc) (host port 2383)
==> virtualbox-iso: Executing custom VBoxManage commands...
    virtualbox-iso: Executing: modifyvm foo --memory 2048
    virtualbox-iso: Executing: modifyvm foo --cpus 1
    virtualbox-iso: Executing: modifyvm foo --audio none
==> virtualbox-iso: Starting the virtual machine...
    virtualbox-iso: The VM will be run headless, without a GUI. If you want to
    virtualbox-iso: view the screen of the VM, connect via VRDP without a password to
    virtualbox-iso: rdp://0.0.0.0:5945
==> virtualbox-iso: Waiting 10s for boot...
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Error sending boot command: VBoxManage error: VBoxManage: error: Could not send all scan codes to the virtual keyboard (VERR_PDM_NO_QUEUE_ITEMS)
==> virtualbox-iso: VBoxManage: error: Details: code VBOX_E_IPRT_ERROR (0x80bb0005), component KeyboardWrap, interface IKeyboard, callee nsISupports
==> virtualbox-iso: VBoxManage: error: Context: "PutScancodes(ComSafeArrayAsInParam(saScancodes), &codesStored)" at line 345 of file VBoxManageControlVM.cpp
==> virtualbox-iso: Unregistering and deleting virtual machine...
==> virtualbox-iso: Error deleting virtual machine: VBoxManage error: VBoxManage: error: Cannot unregister the machine 'foo' while it is locked
==> virtualbox-iso: VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component MachineWrap, interface IMachine, callee nsISupports
==> virtualbox-iso: VBoxManage: error: Context: "Unregister(CleanupMode_DetachAllReturnHardDisksOnly, ComSafeArrayAsOutParam(aMedia))" at line 156 of file VBoxManageMisc.cpp
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Error sending boot command: VBoxManage error: VBoxManage: error: Could not send all scan codes to the virtual keyboard (VERR_PDM_NO_QUEUE_ITEMS)
VBoxManage: error: Details: code VBOX_E_IPRT_ERROR (0x80bb0005), component KeyboardWrap, interface IKeyboard, callee nsISupports
VBoxManage: error: Context: "PutScancodes(ComSafeArrayAsInParam(saScancodes), &codesStored)" at line 345 of file VBoxManageControlVM.cpp

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Error sending boot command: VBoxManage error: VBoxManage: error: Could not send all scan codes to the virtual keyboard (VERR_PDM_NO_QUEUE_ITEMS)
VBoxManage: error: Details: code VBOX_E_IPRT_ERROR (0x80bb0005), component KeyboardWrap, interface IKeyboard, callee nsISupports
VBoxManage: error: Context: "PutScancodes(ComSafeArrayAsInParam(saScancodes), &codesStored)" at line 345 of file VBoxManageControlVM.cpp

==> Builds finished but no artifacts were created.
```

```
[packer@localhost ~]$ vboxmanage showvminfo --machinereadable foo | grep "VMState="
VMState="gurumeditation"
```

**Comment by pstrzelczak** [comment]
I reproduced this issue with both packer and virtualbox debug logs enabled:

[packer.txt](https://github.com/hashicorp/packer/files/1126956/packer.txt)
[VBoxSVC.txt](https://github.com/hashicorp/packer/files/1126965/VBoxSVC.txt)

**Comment by pstrzelczak** [comment]
I reported the original problem to Oracle VirtualBox: https://www.virtualbox.org/ticket/16879

**Comment by arizvisa** [comment]
Does this issue still occur? It looks like it was closed by #5512

**Comment by sylviamoss** [comment]
Hi @pstrzelczak!
We believe the issue was fixed by #5512 as @arizvisa said. I will close it for now and if you still think this is an issue for you, please feel free to report that to us. 😄 
Thanks!
